src/backend/curses.rs

```diff
@@ -0,0 +1,231 @@
+use std::io;
+
+use crate::backend::Backend;
+use crate::buffer::Cell;
+use crate::layout::Rect;
+use crate::style::{Color, Modifier, Style};
+#[cfg(unix)]
+use crate::symbols::{bar, block, line, DOT};
+#[cfg(unix)]
+use pancurses::ToChtype;
+#[cfg(unix)]
+use unicode_segmentation::UnicodeSegmentation;
+
+pub struct CursesBackend {
+    curses: easycurses::EasyCurses,
+}
+
+impl CursesBackend {
+    pub fn new() -> Result<CursesBackend, String> {
+        match easycurses::EasyCurses::initialize_system() {
```

Comment:
Unless part of this setup is required for this backend to work I would rather have the user give the constructor a configured curses instance instead of doing the configuration internally. I have already experimented with this in the Termion backend and I have found that it is very difficult to provide a configuration that match all use cases.

Reply:
That makes sense, I've kept `new()` as a convenience constructor and added a `with_curses()` method that takes a configured instance. This is in line with how the rustbox and crossterm backends handle things.